### PR TITLE
1.2: backports 18-12-03

### PR DESCRIPTION
**Pull request by ianvernon**
Backported PRs:

* #6346 
* #6318 

Signed-off by: Ian Vernon <ian@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6361)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-missed-k8s

**Comment by ianvernon** [comment]
test-docs-please

**Comment by ianvernon** [comment]
Unit tests failed due to legitimate error:

```
03:15:04 fatal error: sync: Unlock of unlocked RWMutex
03:15:04 
03:15:04 goroutine 309 [running]:
03:15:04 runtime.throw(0x28fb778, 0x20)
03:15:04 	/usr/local/go/src/runtime/panic.go:616 +0x81 fp=0xc421755450 sp=0xc421755430 pc=0xf5dbc1
03:15:04 sync.throw(0x28fb778, 0x20)
03:15:04 	/usr/local/go/src/runtime/panic.go:605 +0x35 fp=0xc421755470 sp=0xc421755450 pc=0xf5db35
03:15:04 sync.(*RWMutex).Unlock(0xc4202026c4)
03:15:04 	/usr/local/go/src/sync/rwmutex.go:125 +0xa1 fp=0xc4217554a0 sp=0xc421755470 pc=0xf962b1
03:15:04 github.com/cilium/cilium/pkg/endpoint.(*Endpoint).Unlock(0xc4202026c0)
03:15:04 	/go/src/github.com/cilium/cilium/pkg/endpoint/endpoint.go:2728 +0x31 fp=0xc4217554b8 sp=0xc4217554a0 pc=0x1f3d7e1
03:15:04 github.com/cilium/cilium/pkg/endpoint.(*Endpoint).runLabelsResolver(0xc4202026c0, 0x2aea500, 0xc4202b0640, 0x1, 0xc421755701)
03:15:04 	/go/src/github.com/cilium/cilium/pkg/endpoint/endpoint.go:2374 +0xa9 fp=0xc421755588 sp=0xc4217554b8 pc=0x1f3ae79
03:15:04 github.com/cilium/cilium/pkg/endpoint.(*Endpoint).UpdateLabels(0xc4202026c0, 0x2aea500, 0xc4202b0640, 0xc421755870, 0xc422bc9170, 0x1)
03:15:04 	/go/src/github.com/cilium/cilium/pkg/endpoint/endpoint.go:2350 +0x47b fp=0xc421755768 sp=0xc421755588 pc=0x1f3ad4b
03:15:04 github.com/cilium/cilium/daemon.(*Daemon).createEndpoint(0xc4202b0640, 0xc4201ee8c0, 0xc4203d057c, 0x4, 0x0, 0x0, 0x0, 0x7efdc457d8f0, 0x1b, 0x2af4820)
03:15:04 	/go/src/github.com/cilium/cilium/daemon/endpoint.go:207 +0x269 fp=0xc421755980 sp=0xc421755768 pc=0x231f8f9
03:15:04 github.com/cilium/cilium/daemon.(*DaemonSuite).TestEndpointAddNoLabels(0xc4200f0000, 0xc4201f85a0)
03:15:04 	/go/src/github.com/cilium/cilium/daemon/endpoint_test.go:74 +0x98 fp=0xc421755b60 sp=0xc421755980 pc=0x2353038
03:15:04 runtime.call32(0xc4218d8f60, 0xc4218de0d8, 0xc4218e0dd0, 0x1000000010)
03:15:04 	/usr/local/go/src/runtime/asm_amd64.s:573 +0x3b fp=0xc421755b90 sp=0xc421755b60 pc=0xf8a91b
03:15:04 reflect.Value.call(0x2875920, 0xc4200f0000, 0x4e13, 0x28b995a, 0x4, 0xc420088f70, 0x1, 0x1, 0x3f639c0, 0x2889440, ...)
03:15:04 	/usr/local/go/src/reflect/value.go:447 +0x969 fp=0xc421755e50 sp=0xc421755b90 pc=0xfeb4f9
03:15:04 reflect.Value.Call(0x2875920, 0xc4200f0000, 0x4e13, 0xc420088f70, 0x1, 0x1, 0xc4201f8690, 0xc420088f07, 0x14b8140)
03:15:04 	/usr/local/go/src/reflect/value.go:308 +0xa4 fp=0xc421755eb8 sp=0xc421755e50 pc=0xfeaa74
03:15:04 github.com/cilium/cilium/vendor/gopkg.in/check%2ev1.(*suiteRunner).forkTest.func1(0xc4201f85a0)
03:15:04 	/go/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:772 +0x575 fp=0xc421755f98 sp=0xc421755eb8 pc=0x14ba445
03:15:04 github.com/cilium/cilium/vendor/gopkg.in/check%2ev1.(*suiteRunner).forkCall.func1(0xc4202e0400, 0xc4201f85a0, 0xc4215c6d80)
03:15:04 	/go/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:666 +0x7c fp=0xc421755fc8 sp=0xc421755f98 pc=0x14b9d8c
03:15:04 runtime.goexit()
03:15:04 	/usr/local/go/src/runtime/asm_amd64.s:2361 +0x1 fp=0xc421755fd0 sp=0xc421755fc8 pc=0xf8d0e1
03:15:04 created by github.com/cilium/cilium/vendor/gopkg.in/check%2ev1.(*suiteRunner).forkCall
03:15:04 	/go/src/github.com/cilium/cilium/vendor/gopkg.in/check.v1/check.go:663 +0x20c
```

I'll track down the issue. Moving back to `wip` state accordingly.

File: pkg/endpoint/endpoint.go [hunk 1]
```diff
@@ -2368,18 +2368,15 @@ func (e *Endpoint) identityResolutionIsObsolete(myChangeRev int) bool {
 }
 
 // Must be called with e.Mutex NOT held.
-func (e *Endpoint) runLabelsResolver(owner Owner, myChangeRev int) {
-	// NOTE: UnconditionalLock is used here only for logging
-	e.UnconditionalLock()
```
Comment (aanm) [inline pkg/endpoint/endpoint.go]: @ianvernon This change is not in any of the upstream PRs

**Review by aanm — DISMISSED** [review]

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-missed-k8s

**Comment by ianvernon** [comment]
test-docs-please

File: pkg/endpoint/endpoint.go (hunk 1 above)
Comment (ianvernon) [reply pkg/endpoint/endpoint.go]: Fixed.

**Comment by ianvernon** [comment]
Sigh, hit `go fmt` failures due to differences between 1.10 and 1.11: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/8401/execution/node/91/log/?consoleFull

Will fix and reupload...

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-missed-k8s

**Comment by ianvernon** [comment]
K8s 1.7 tests failed for Kafka:

```
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
Failed to produce to empire-hq on topic empire-announce
Expected
    <*errors.errorString | 0xc420e023c0>: {
        s: "ExecKafkaPodCmd: command 'kubectl exec -n default empire-hq-2917507631-67n7b sh -- -c \"echo 'Happy 40th Birthday to General Tagge' | ./kafka-produce.sh --topic empire-announce\"' failed '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' || '[2018-12-03 20:00:17,738] WARN Removing server kafka-service:9092 from bootstrap.servers as DNS resolution failed for kafka-service (org.apache.kafka.clients.ClientUtils)\norg.apache.kafka.common.KafkaException: Failed to construct kafka producer\n\tat org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:416)\n\tat org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:288)\n\tat kafka.producer.NewShinyProducer.<init>(BaseProducer.scala:40)\n\tat kafka.tools.ConsoleProducer$.main(ConsoleProducer.scala:48)\n\tat kafka.tools.ConsoleProducer.main(ConsoleProducer.scala)\nCaused by: org.apache.kafka.common.config.ConfigException: No resolvable bootstrap urls given in bootstrap.servers\n\tat org.apache.kafka.clients.ClientUtils.parseAndValidateAddresses(ClientUtils.java:64)\n\tat org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:373)\n\t... 4 more\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'",
    }
to be nil
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/k8sT/KafkaPolicies.go:169
```

Link: 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/8405

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-docs-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**